Jellyfin.Server/Migrations/Routines/MigrateKeyframeData.cs

```diff
@@ -63,48 +56,36 @@ public MigrateKeyframeData(
     /// <inheritdoc />
     public void Perform()
     {
-        const int Limit = 100;
-        int itemCount = 0, offset = 0, previousCount;
+        const int Limit = 5000;
+        int itemCount = 0, offset = 0;
 
         var sw = Stopwatch.StartNew();
-        var itemsQuery = new InternalItemsQuery
-        {
-            MediaTypes = [MediaType.Video],
-            SourceTypes = [SourceType.Library],
-            IsVirtualItem = false,
-            IsFolder = false
-        };
 
         using var context = _dbProvider.CreateDbContext();
+        var records = context.BaseItems.Where(b => b.MediaType == MediaType.Video.ToString() && !b.IsVirtualItem).Count();
+        _logger.LogInformation("Checking {Count} items for importable keyframe data.", records);
+
         context.KeyframeData.ExecuteDelete();
         using var transaction = context.Database.BeginTransaction();
-        List<KeyframeData> keyframes = [];
-
         do
         {
-            var result = _libraryManager.GetItemsResult(itemsQuery);
-            _logger.LogInformation("Importing keyframes for {Count} items", result.TotalRecordCount);
-
-            var items = result.Items;
-            previousCount = items.Count;
-            offset += Limit;
+            var items = context.BaseItems.Where(b => b.MediaType == MediaType.Video.ToString() && !b.IsVirtualItem).OrderBy(e => e.Id).Skip(offset).Take(Limit).ToList();
             foreach (var item in items)
             {
                 if (TryGetKeyframeData(item, out var data))
                 {
-                    keyframes.Add(data);
+                    itemCount++;
+                    context.KeyframeData.Add(data);
                 }
+            }
 
-                if (++itemCount % 10_000 == 0)
-                {
-                    context.KeyframeData.AddRange(keyframes);
-                    keyframes.Clear();
-                    _logger.LogInformation("Imported keyframes for {Count} items in {Time}", itemCount, sw.Elapsed);
-                }
+            offset += items.Count;
+            if (offset % 5_000 == 0)
```

Comment:
This check can be removed to always log since the limit is also 5k (repeat for other routines updated)